### PR TITLE
feature : add injection with koin

### PR DESCRIPTION
**Pull request by josemiZ**
We need to be cautious with view models, because I made some changes in the constructor

### CODE REVIEW

File: app/src/main/java/com/anwera64/pagodividido/domain/repository/TripRepository.kt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.anwera64.pagodividido.domain.repository
 
 import androidx.annotation.WorkerThread
+import com.anwera64.pagodividido.data.AppDatabase
```
Comment (Anwera64) [inline app/src/main/java/com/anwera64/pagodividido/domain/repository/TripRepository.kt:4]: Remove unused import

File: app/src/main/java/com/anwera64/pagodividido/presentation/base/BaseViewModelFragment.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+abstract class BaseViewModelFragment<V : ViewModel, B : ViewDataBinding>(type: KClass<V>) : BaseFragment<B>() {
+
+    protected val viewModel: V by currentScope.viewModel(this, type)
+    abstract val viewModelValue: Int?
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModelValue?.let {
+            binding.setVariable(it, viewModel)
```
Comment (Anwera64) [inline app/src/main/java/com/anwera64/pagodividido/presentation/base/BaseViewModelFragment.kt]: add name to it variable

**Review by Anwera64 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/anwera64/pagodividido/presentation/base/BaseViewModelFragment.kt (hunk 2 above)
Comment (josemiZ) [reply app/src/main/java/com/anwera64/pagodividido/presentation/base/BaseViewModelFragment.kt]: done in base fragment and also in base activity

File: app/src/main/java/com/anwera64/pagodividido/domain/repository/TripRepository.kt (hunk 1 above)
Comment (josemiZ) [reply app/src/main/java/com/anwera64/pagodividido/domain/repository/TripRepository.kt:4]: done

**Pull request merged**